packages/components/input/src/use-input.ts

```diff
@@ -250,17 +250,29 @@ export function useInput<T extends HTMLInputElement | HTMLTextAreaElement = HTML
   const hasPlaceholder = !!props.placeholder;
   const hasLabel = !!label;
   const hasHelper = !!description || !!errorMessage;
-  const shouldLabelBeOutside = labelPlacement === "outside" || labelPlacement === "outside-left";
+
+  /*
+    outside -> label is outside only when some placeholder is there
+    outside-top, outside-left-> label is outside regardless of placeholder
+  */
+  const shouldLabelBeOutside =
+    labelPlacement === "outside" ||
+    labelPlacement === "outside-left" ||
+    labelPlacement === "outside-top";
   const shouldLabelBeInside = labelPlacement === "inside";
   const isPlaceholderShown = domRef.current
     ? (!domRef.current.value || domRef.current.value === "" || !inputValue || inputValue === "") &&
       hasPlaceholder
     : false;
+
   const isOutsideLeft = labelPlacement === "outside-left";
+  const isOutsideTop = labelPlacement === "outside-top";
 
   const hasStartContent = !!startContent;
+
   const isLabelOutside = shouldLabelBeOutside
     ? labelPlacement === "outside-left" ||
+      labelPlacement === "outside-top" ||
```

Comment:
> 2. also i got `This comparison appears to be unintentional because the types '"inside" | undefined' and '"outside-top"' have no overlap.ts(2367)`. please double check.

I don'y see this message in IDE or chrome dev tools. can you tell me where you got this?

Reply:
> it looks kinda packed. probably put outsied-left and outside-top in a new row. please also check with docs one also. 

is this okay?
![image](https://github.com/user-attachments/assets/b8c5c1bc-bfef-4b3e-b4aa-50972d2f6d4c)

Reply:
looks good